### PR TITLE
Add keychron k17 pro

### PR DESCRIPTION
**Pull request by KeychronMacro**
As the title and request for a review, thanks!

## Description

- Added k17_pro/ansi_encoder rgb and white version
- Added q10_pro/ansi_encoder and q10_pro/iso_encoder
- Added q13_pro/ansi_encoder and q13_pro/iso_encoder

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Mchh85** [comment]
@keychron-dev @lokher 您好，我买的q13 pro 已经到手了，我需要修改固件。什么时候在bluetooth_playground分支中合并这个啊？已经买了五天了，一直没有这个型号。

**Comment by lokher** [comment]
> @keychron-dev @lokher 您好，我买的q13 pro 已经到手了，我需要修改固件。什么时候在bluetooth_playground分支中合并这个啊？已经买了五天了，一直没有这个型号。

我们最近比较忙，要晚点才能处理，你可以直接从以下分支把代码提取出来放到 `bluetooth_playgroun` 中编译
https://github.com/lalalademaxiya1/qmk_firmware/tree/keychron_k17_pro

File: keyboards/keychron/bluetooth/bluetooth.c [hunk 1]
```diff
@@ -291,7 +292,16 @@ uint8_t bluetooth_keyboard_leds(void) {
 extern keymap_config_t keymap_config;
 
 void bluetooth_send_keyboard(report_keyboard_t *report) {
-    if (bt_state == BLUETOOTH_PARING && !pincodeEntry) return;
+    /* Prevent keys from not bouncing back after shutdown due to low power */
```
Comment (lokher) [inline keyboards/keychron/bluetooth/bluetooth.c]:
```suggestion
    /* Release all keys before  low battery shutting down */
```

File: keyboards/keychron/k17_pro/k17_pro.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    BT_HST2,
+    BT_HST3,
+    BAT_LVL,
+#else
+    BT_HST1 = KC_TRNS,
+    BT_HST2 = KC_TRNS,
+    BT_HST3 = KC_TRNS,
+    BAT_LVL = KC_TRNS,
+#endif
+	NEW_SAFE_RANGE
```
Comment (lokher) [inline keyboards/keychron/k17_pro/k17_pro.h]:
```suggestion
	NEW_SAFE_RANGE
```
```suggestion
    NEW_SAFE_RANGE
```

File: keyboards/keychron/q10_pro/ansi_encoder/ansi_encoder.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
+/* Copyright 2022 @ Keychron (https://www.keychron.com)
```
Comment (lokher) [inline keyboards/keychron/q10_pro/ansi_encoder/ansi_encoder.c]:
```suggestion
/* Copyright 2023 @ Keychron (https://www.keychron.com)
```

File: keyboards/keychron/q10_pro/ansi_encoder/config.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/* Copyright 2022 @ Keychron (https://www.keychron.com)
```
Comment (lokher) [inline keyboards/keychron/q10_pro/ansi_encoder/config.h]:
```suggestion
/* Copyright 2023 @ Keychron (https://www.keychron.com)
```

File: keyboards/keychron/q10_pro/config.h [hunk 5]
```diff
@@ -0,0 +1,83 @@
+/* Copyright 2022 @ Keychron(https://www.keychron.com)
```
Comment (lokher) [inline keyboards/keychron/q10_pro/config.h]:
```suggestion
/* Copyright 2023 @ Keychron(https://www.keychron.com)
```

File: keyboards/keychron/q10_pro/iso_encoder/config.h [hunk 6]
```diff
@@ -0,0 +1,63 @@
+/* Copyright 2022 @ Keychron (https://www.keychron.com)
```
Comment (lokher) [inline keyboards/keychron/q10_pro/iso_encoder/config.h]:
```suggestion
/* Copyright 2023 @ Keychron (https://www.keychron.com)
```

File: keyboards/keychron/q10_pro/iso_encoder/iso_encoder.c [hunk 7]
```diff
@@ -0,0 +1,124 @@
+/* Copyright 2022 @ Keychron (https://www.keychron.com)
```
Comment (lokher) [inline keyboards/keychron/q10_pro/iso_encoder/iso_encoder.c]:
```suggestion
/* Copyright 2023 @ Keychron (https://www.keychron.com)
```

File: keyboards/keychron/q13_pro/ansi_encoder/ansi_encoder.c [hunk 8]
```diff
@@ -0,0 +1,140 @@
+/* Copyright 2022 @ Keychron (https://www.keychron.com)
```
Comment (lokher) [inline keyboards/keychron/q13_pro/ansi_encoder/ansi_encoder.c]:
```suggestion
/* Copyright 2023 @ Keychron (https://www.keychron.com)
```

File: keyboards/keychron/q13_pro/iso_encoder/iso_encoder.c [hunk 9]
```diff
@@ -0,0 +1,141 @@
+/* Copyright 2022 @ Keychron (https://www.keychron.com)
```
Comment (lokher) [inline keyboards/keychron/q13_pro/iso_encoder/iso_encoder.c]:
```suggestion
/* Copyright 2023 @ Keychron (https://www.keychron.com)
```

**Review by lokher — CHANGES_REQUESTED** [review]

**Review by lokher — APPROVED** [review]

**Comment by campbellhennessy** [comment]
Hello, I am wondering when this PR is planned to be merged into the bluetooth_playground branch?

**Pull request closed without merging**